crates/amaru/src/bin/amaru/cmd/daemon.rs

```diff
@@ -70,9 +70,15 @@ pub async fn run(
 
     let mut clients: Vec<(String, Arc<Mutex<PeerClient>>)> = vec![];
     for peer in &config.upstream_peers {
-        let client =
-            PeerClient::connect(peer.clone(), config.network.to_network_magic() as u64).await?;
-        clients.push((peer.clone(), Arc::new(Mutex::new(client))));
+        match PeerClient::connect(peer.clone(), config.network.to_network_magic() as u64).await {
```

Comment:
Another option that is a bit less verbose would be to chain `.await.inspect_err(|e| tracing::error!("failed to connect to peer {}: {}", peer, e)?;`

Reply:
Thanks for the feedback, greatly appreciated as I may often miss a rust idioms.

Done.